extensions/csrf-reactive/runtime/src/main/java/io/quarkus/csrf/reactive/runtime/CsrfRequestResponseReactiveFilter.java

```diff
@@ -71,11 +72,12 @@ public Uni<Response> filter(ContainerRequestContext requestContext, RoutingConte
             routing.put(CSRF_TOKEN_KEY, cookieToken);
 
             try {
-                int suppliedTokenSize = Base64.getUrlDecoder().decode(cookieToken).length;
-
-                if (suppliedTokenSize != config.tokenSize) {
-                    LOG.debugf("Invalid CSRF token cookie size: expected %d, got %d", config.tokenSize,
-                            suppliedTokenSize);
+                int cookieTokenSize = Base64.getUrlDecoder().decode(cookieToken).length;
+                // HMAC SHA256 output is 32 bytes long
+                int expectedCookieTokenSize = config.tokenSignatureKey.isPresent() ? 32 : config.tokenSize;
+                if (cookieTokenSize != expectedCookieTokenSize) {
+                    LOG.debugf("Invalid CSRF token cookie size: expected %d, got %d", expectedCookieTokenSize,
```

Comment:
Shouldn't an exception be thrown here ?

Reply:
@gastaldi See the next line, `return Uni.createFrom().item(badClientRequest());` - leads to `401`